packages/shared/src/deprecated.ts

```diff
@@ -28,7 +29,7 @@ export const deprecated = (fnName: string, warning: string, key?: string): void
   }
   displayedWarnings.add(messageId);
 
-  console.warn(
+  logger.warnOnce(
```

Comment:
This change isn't correct, we'd be tracking the message in two separate sets. The functionality of `warnOnce` is already given from the code above this code block. If you don't want to combine the `deprecated` function and the new logger in this PR, let's leave it for now and it can be refactored later

Reply:
Ah yeah I didn't think that's what you meant and I missed that. I'll leave it for another PR.